library/ssl_tls13_generic.c

```diff
@@ -2944,14 +2944,14 @@ static int ssl_new_session_ticket_parse( mbedtls_ssl_context* ssl,
 
         memcpy( ssl->session->ticket_nonce, &buf[9], ssl->session->ticket_nonce_len );
 
-        MBEDTLS_SSL_DEBUG_BUF( 3, "ticket->nonce:", (unsigned char*)&buf[9],
-        ssl->session->ticket_nonce_len );
+        MBEDTLS_SSL_DEBUG_BUF( 3, "ticket->nonce:", &buf[9],
+                ssl->session->ticket_nonce_len );
 
     }
 
     /* Ticket */
-    ticket_len = ( buf[9 + ssl->session->ticket_nonce_len] << 8 ) |
-                 ( buf[10 + ssl->session->ticket_nonce_len] );
+    ticket_len = ( (unsigned) buf[9 + ssl->session->ticket_nonce_len] << 8 ) |
```

Comment:
ticket_len is `uint16_t`, should we use that instead?

Reply:
I just follow L2967, the `ext_len` is a `uint16_t` too.
```
ext_len = ( (unsigned) buf[ 11 + ssl->session->ticket_nonce_len + ticket_len ] << 8 ) |
```

Reply:
Hm, I'm not sure how we handle this throughout the library, but generally I'd prefer to use `size_t` here in order to avoid overflow bugs resulting from subsequent modification of the parsed variable. The Fizz CVE https://github.com/facebookincubator/fizz/commit/40bbb161e72fb609608d53b9d64c56bb961a6ee2 comes to mind...

Reply:
See also https://github.com/ARMmbed/mbedtls/pull/3238#pullrequestreview-411642455. For now, I think we should use `size_t` for all local variables holding length values.